source/application/Utils/File.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2016- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#pragma once
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace Utils
+{
+    namespace File
+    {
+        class File

```

Comment:
The DirLocks functionality, which will want to create a temp-file with "delete-on-close".